presto-hive-hadoop2/src/test/java/com/facebook/presto/hive/s3select/TestHiveFileSystemS3SelectJsonPushdown.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.s3select;
+
+import com.facebook.presto.hive.HiveColumnHandle;
+import com.facebook.presto.spi.ColumnHandle;
+import com.facebook.presto.spi.SchemaTableName;
+import com.facebook.presto.testing.MaterializedResult;
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Parameters;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.airlift.testing.Assertions.assertEqualsIgnoreOrder;
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.hive.HiveColumnHandle.ColumnType.REGULAR;
+import static com.facebook.presto.hive.HiveFileSystemTestUtils.filterTable;
+import static com.facebook.presto.hive.HiveFileSystemTestUtils.newSession;
+import static com.facebook.presto.hive.HiveFileSystemTestUtils.readTable;
+import static com.facebook.presto.hive.HiveType.HIVE_LONG;
+
+public class TestHiveFileSystemS3SelectJsonPushdown
+{
+    private S3SelectTestHelper s3SelectTestHelper;
+    private SchemaTableName jsonTable;
+
+    @Parameters({
+            "hive.hadoop2.metastoreHost",
+            "hive.hadoop2.metastorePort",
+            "hive.hadoop2.databaseName",
+            "hive.hadoop2.s3.awsAccessKey",
+            "hive.hadoop2.s3.awsSecretKey",
+            "hive.hadoop2.s3.writableBucket"})
+    @BeforeClass
+    public void setup(String host, int port, String databaseName, String awsAccessKey, String awsSecretKey, String writableBucket)
+    {
+        s3SelectTestHelper = new S3SelectTestHelper(host, port, databaseName, awsAccessKey, awsSecretKey, writableBucket, true);
```

Comment:
Missing `s3SelectTestHelper.setUp()` call. Does this test pass in the current state?

Reply:
Yes, the test passes:
<img width="1776" alt="image" src="https://user-images.githubusercontent.com/9262928/214357620-471b982c-b4b2-4a67-801e-9f9b0c9a7a01.png">

Reply:
The executor is also instantiated in the constructor and that is actually passed on as a parameter, so it did not depend on the setup method, it just does not release the resources explicitly.

Reply:
I ditched the setup method in utility class and updated the tests teardown. Nice catch! 🥇